From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Davis Goodin <dagood@microsoft.com>
Date: Thu, 5 Oct 2023 16:06:31 -0700
Subject: [PATCH] Skip failing GDB tests on Windows

---
 src/runtime/runtime-gdb_test.go | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/runtime/runtime-gdb_test.go b/src/runtime/runtime-gdb_test.go
index 8c759bf4066370..5249f87154245f 100644
--- a/src/runtime/runtime-gdb_test.go
+++ b/src/runtime/runtime-gdb_test.go
@@ -168,6 +168,9 @@ func TestGdbPythonCgo(t *testing.T) {
 }
 
 func testGdbPython(t *testing.T, cgo bool) {
+	if runtime.GOOS == "windows" {
+		t.Skip("may encounter unexpected line endings")
+	}
 	if cgo {
 		testenv.MustHaveCGO(t)
 	}
@@ -399,6 +402,9 @@ func main() {
 // TestGdbBacktrace tests that gdb can unwind the stack correctly
 // using only the DWARF debug info.
 func TestGdbBacktrace(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("may encounter unexpected line endings")
+	}
 	if runtime.GOOS == "netbsd" {
 		testenv.SkipFlaky(t, 15603)
 	}
@@ -525,6 +531,9 @@ func main() {
 // TestGdbAutotmpTypes ensures that types of autotmp variables appear in .debug_info
 // See bug #17830.
 func TestGdbAutotmpTypes(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("may encounter unexpected line endings")
+	}
 	checkGdbEnvironment(t)
 	t.Parallel()
 	checkGdbVersion(t)
@@ -598,6 +607,9 @@ func main() {
 `
 
 func TestGdbConst(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("may encounter unexpected line endings")
+	}
 	checkGdbEnvironment(t)
 	t.Parallel()
 	checkGdbVersion(t)
