From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: bot-for-go[bot] <199222863+bot-for-go[bot]@users.noreply.github.com>
Date: Thu, 7 May 2026 16:14:34 -0700
Subject: [PATCH] Skip rerunWithFIPS140Enforced

---
 src/crypto/tls/fips140_test.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/crypto/tls/fips140_test.go b/src/crypto/tls/fips140_test.go
index bbaa178c9dac45..4a4ec4352d4446 100644
--- a/src/crypto/tls/fips140_test.go
+++ b/src/crypto/tls/fips140_test.go
@@ -60,6 +60,7 @@ func generateKeyShare(group CurveID) keyShare {
 
 func rerunWithFIPS140Enforced(t *testing.T) {
 	t.Helper()
+	t.Skip("remainder of test requires FIPS 140 mode; this is not necessarily possible on all systems when using systemcrypto (Microsoft build of Go)")
 	if err := ifips140.Supported(); err != nil {
 		t.Skipf("test requires FIPS 140 mode: %v", err)
 	}
